File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.cpp
class Solution {
public:
    bool check(int x){
        return (x % 2 == 1) ? true : false;
    }
    
    string largestOddNumber(string num) {
        
        string res = "";
        
        int i = num.size() - 1;
        
        while(i >= 0 && !check(num[i] - '0'))   --i;
        
        if(i < 0)   return res;
        
        return num.substr(0, i + 1);
        
    }
};

File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String num) {
        
        int i = num.length() - 1;
        
        while(i >= 0){
            
            if((num.charAt(i) - '0') % 2 == 1)  return num.substring(0, i + 1);
            
            --i;
            
        }
        
        return new String("");
        
    }
}